test/jdk/com/sun/java/swing/plaf/gtk/bug6492108.java

```diff
@@ -0,0 +1,155 @@
+
+    private JPanel panel;
+
+    public static void main(String[] args) throws Throwable {
+        try {
+            UIManager.setLookAndFeel(
+                "com.sun.java.swing.plaf.gtk.GTKLookAndFeel");
+        } catch (Exception e) {
+            throw new SkippedException("GTK LAF is not supported on this system;" +
+                    " test passes");
+        }
+        new bug6492108().run(args);
+    }
+
+    private static void addTextComps(Container parent,
+                                     Class<? extends JTextComponent> type)
+            throws Throwable
+    {
+        JTextComponent text = type.newInstance();
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEditable(false);
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        text.setEditable(false);
+        addTextComp(parent, text);
+    }
+
+    private static void addTextComp(Container parent, JTextComponent text) {
+        JScrollPane sp = new JScrollPane(text);
+        text.setFocusable(false); // to avoid showing the blinking caret
+        sp.setPreferredSize(new Dimension(150, 150));
+        sp.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        sp.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_NEVER);
+        parent.add(sp);
+    }
+
+    protected Component createContentPane() {
+        panel = new JPanel();
+        panel.setLayout(new GridLayout(3, 4));
+        try {
+            addTextComps(panel, JTextArea.class);
+            addTextComps(panel, JEditorPane.class);
+            addTextComps(panel, JTextPane.class);
+        } catch (Throwable t) {
+            fail("Problem creating text components");
+        }
+        return panel;
+    }
+
+    private void onEDT10() {
+        requestAndWaitForFocus(panel);
+    }
+
+    private void onBackgroundThread20() {
+        // For each component on the top row, compare against the two
+        // components below in the same column.  All three components in
+        // that column should be the same pixel-for-pixel.
+        for (int count = 0; count < 4; count++) {
+            Component ref = panel.getComponent(count);
+            Point loc = ref.getLocationOnScreen();
+            Rectangle refRect =
+                new Rectangle(loc.x, loc.y, ref.getWidth(), ref.getHeight());
+            BufferedImage refimg = robot.createScreenCapture(refRect);
+
+            for (int k = 1; k < 3; k++) {
+                int index = count + (k*4);
+                Component test = panel.getComponent(index);
+                loc = test.getLocationOnScreen();
+                Rectangle testRect =
+                    new Rectangle(loc.x, loc.y,
+                                  test.getWidth(), test.getHeight());
```

Comment:
```suggestion
                loc = test.getLocationOnScreen();
                Rectangle testRect = new Rectangle(test.getLocationOnScreen(),
                                                   test.getSize());
```
Choose one style and adjust both statements, `refRect` and `testRect`.

Reply:
Updated to use same style for both places.